Bump @nextcloud/vue from 1.5.0 to 2.2.1

georgehrke opened the pull request:
Signed-off-by: Georg Ehrke <developer@georgehrke.com>

Breaking changes:
- [x] `#app-content` to `.app-content`
- [x] `#app-navigation` to `.app-navigation`
- [x] `#app-sidebar` to `.app-sidebar`
- [x]  `#content` to `.content`
- [x] Navigation List Slot
- [x] AppNavigationSettings Footer Slot

georgehrke committed:
Bump @nextcloud/vue from 1.5.0 to 2.2.1

Signed-off-by: Georg Ehrke <developer@georgehrke.com>

georgehrke committed:
Fix breaking changes of nc/vue update

Signed-off-by: Georg Ehrke <developer@georgehrke.com>

georgehrke commented:
/compile

georgehrke commented:
/compile /

skjnldsv commented on apps/settings/src/views/Users.vue:
```diff
@@ -92,58 +92,60 @@
-				</div>
-			</AppNavigationSettings>
+			</template>
+			<template #footer>
```
Wait, why did we need a footer slot? I completely missed this! :thinking:

georgehrke replied:
https://github.com/nextcloud/nextcloud-vue/releases/tag/v2.2.0 😉

skjnldsv replied:
Interesting choice, I would have put the default slot to be the list and not the Navigation `new` button :shrug:

skjnldsv reviewed (approved):
Thanks for this @georgehrke :heart:

skjnldsv commented:
Needs big rebase.

georgehrke commented:
Rebased

ChristophWurst commented:
> dependageorg

Nice

ChristophWurst reviewed (approved):
Not sure if ready to review but the code looks fine :)

skjnldsv commented:
Acceptance apps/users fails & jsunit too.

georgehrke commented:
```
PhantomJS 2.1.1 (Linux 0.0.0) DEBUG: 'OCA.Files.Settings initialized'
1011	PhantomJS 2.1.1 (Linux 0.0.0) ERROR
1012	  {
1013	    "message": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/personal-settings.js:1:0",
1014	    "str": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/personal-settings.js:1:0"
1015	  }
1016	PhantomJS 2.1.1 (Linux 0.0.0): Executed 0 of 0 ERROR (0.374 secs / 0 secs)
1017	PhantomJS 2.1.1 (Linux 0.0.0) ERROR
1018	  {
1019	    "message": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/sidebar.js:1:0",
1020	    "str": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/sidebar.js:1:0"
1021	  }
```

skjnldsv commented:
I guess this is the same issue I reported recently.
There is an issue loading the scripts on files on master :thinking:

georgehrke commented:
> I guess this is the same issue I reported recently.
There is an issue loading the scripts on files on master 🤔

The nc/vue bundle seems to be using const in the built version. I added it to the webpack config.

skjnldsv commented:
> Transpile @nc/vue and some of its dependencies 2e6775b7e807609b531c66030b9c9b7e13be4b6b

I'm not sure abou tthis one! Don't we already transpile on the library?
https://github.com/nextcloud/nextcloud-vue/blob/master/babel.config.js

georgehrke committed:
Recompile JS assets

Signed-off-by: Georg Ehrke <developer@georgehrke.com>

danxuliu committed:
Hide share link menu before copying the share link in acceptance tests

The (old) Firefox version used in the acceptance tests does not properly
render the share link menu. As the menu is taller than it should
sometimes it covers the copy link button, so it is not possible to click
it without hiding the share link menu. Moreover, in those cases the
share menu button is also covered by the share menu, so the menu needs
to be closed by pressing the "Esc" key.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu commented:
I have fixed the share link acceptance tests (basically the share menu was covering in some cases the _Copy link_ button; it should not happen with up to date browsers).

However, while checking this I have found a small issue (latest Firefox):

![Share-Link-Menu-Out-Of-Bounds](https://user-images.githubusercontent.com/26858233/88099668-c6914500-cb9b-11ea-975c-a8afa685b8ba.png)

Since https://github.com/nextcloud/nextcloud-vue/commit/f3b223a54ca5c2b9dc5acb7c131c61d698ba7aaa the Actions menu is always shown to the top if there is enough room (before this was done only for centered menus, so it did not apply to the share menu due to being aligned to the right). However, as can be seen in this case there is not enough room, and yet the menu is still shown to the top (and a scroll bar is not shown either). I do not know if the problem is in the Vue component itself or in the way that it is used in the sharing tab.

skjnldsv commented:
> the problem is in the Vue component itself

this ^

georgehrke commented:
> > the problem is in the Vue component itself
> 
> this ^

That seems to be an edge case with certain view heights.
@skjnldsv @rullzer Are you fine with merging this for now?

rullzer commented:
> > > the problem is in the Vue component itself
> > 
> > 
> > this ^
> 
> That seems to be an edge case with certain view heights.
> @skjnldsv @rullzer Are you fine with merging this for now?

Yep lets do this. But be sure to create an issue so we don't lose track

georgehrke commented:
Created an issue: https://github.com/nextcloud/nextcloud-vue/issues/1215

Let me rebase
